in_toto/models/layout.py

```diff
@@ -126,12 +173,376 @@ def read(data):
     return Layout(**data)
 
 
+  def set_relative_expiration(self, days=0, months=0, years=0):
+    """
+    <Purpose>
+      Set the layout's expiration date in one or more of "days", "months" or
+      "years" from today. If no argument is passed, it defaults to today.
+
+    <Arguments>
+      days:
+              Days from today.
+
+      months:
+              Months from today.
+
+      years:
+              Years from today.
+
+    <Exceptions>
+      securesystemslib.exceptions.FormatError
+              If any of days, months or years is passed and is not an
+              integer.
+
+    """
+    securesystemslib.schema.Integer().check_match(days)
+    securesystemslib.schema.Integer().check_match(months)
+    securesystemslib.schema.Integer().check_match(years)
+
+    self.expires = (datetime.today() + relativedelta(
+        days=days, months=months, years=years)).strftime("%Y-%m-%dT%H:%M:%SZ")
```

Comment:
Do we have a schema for date? can we add a guard?

Reply:
There's a `_validate_expires` method on the Layout, which we could use (it checks schema and date validity). But I'm unit testing `set_relative_expiration` together with `_validate_expires`, so I don't think it is necessary to add a guard here.